### PR TITLE
fix: <Nop> not shown in :map commands

### PR DESCRIPTION
**Pull request by shadmansaleh**
Followup on #16594 
- fix \<Nop> not displayed on `showmap()`
- enhance lua map a internal representation.

### CODE REVIEW

File: src/nvim/getchar.c [hunk 1]
```diff
@@ -3435,7 +3434,8 @@ static void showmap(mapblock_T *mp, bool local)
   size_t len = 1;
 
   if (message_filtered(mp->m_keys)
-      && mp->m_str != NULL && message_filtered(mp->m_str)) {
+      && message_filtered(mp->m_str[0] != NUL ? mp->m_str : (char_u *)"<Nop>")
```
Comment (zeertzjq) [inline src/nvim/getchar.c:3437]: This looks a bit inconsistent, as filtering other `<>` key notations doesn't work.

Comment (shadmansaleh) [reply src/nvim/getchar.c:3437]: I realized that after adding the <Nop>. I was actually thinking of making them filterable too . Should we ?

Comment (zeertzjq) [reply src/nvim/getchar.c:3437]: In `map.txt`:
```
The |:filter| command can be used to select what mappings to list.  The
pattern is matched against the {lhs} and {rhs} in the raw form.
```

Comment (zeertzjq) [reply src/nvim/getchar.c:3437]: I think you can limit this PR to fixing the displaying of rhs `<Nop>` mappings only, and add a test for it.

Comment (shadmansaleh) [reply src/nvim/getchar.c:3437]: You can't generate <Cmd> / <Lua> / <Nop> key in that manner, right ?

**Comment by shadmansaleh** [comment]
Ok I've moved filter related changes to #17423 . Can f5518b78c843e4b187bd246d31bb7fde2925139e stay ?

@zeertzjq I've found another issue here . This https://github.com/neovim/neovim/blob/9ef56961328991e22110c8084cb23e3ef3adde39/src/nvim/getchar.c#L3445 doesn't seem to get triggered when you run `:map ...` . For that showmap writes first map on top of the user inputed commad .

File: src/nvim/getchar.c (hunk 1 above)
Comment (zeertzjq) [reply src/nvim/getchar.c]: For `<Nop>` one can use `:filter /^$/ map`. For `<Cmd>` one can use `:execute`. You haven't added a `<Lua>` notation yet.

**Comment by zeertzjq** [comment]
Vim also seems to have this problem.

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**